### PR TITLE
mention the correct feature gate name for RBD CSI migration

### PR DESCRIPTION
**Pull request by humblec**
The feature gate string (CSIMigrationRBD) was not matching with the
implementation (csiMigrationRBD) and this commit make it in parity

Ref# kubernetes/kubernetes#107571

Master branch has this corrected to `CSIMigrationRBD`  ( https://github.com/kubernetes/kubernetes/pull/107554 ) and `master ` documentation matches.
Release 1.23 has it in current form ie `csiMigrationRBD` and this PR correct the same. 

Signed-off-by: Humble Chirammal <hchiramm@redhat.com>

### CODE REVIEW

**Comment by tengqm** [comment]
How could this happen?!!
/lgtm

**Comment by mehabhalodiya** [comment]
LGTM!

File: content/en/docs/concepts/storage/volumes.md [hunk 1]
```diff
@@ -858,7 +858,7 @@ operations from the existing in-tree plugin to the `rbd.csi.ceph.com` {{<
 glossary_tooltip text="CSI" term_id="csi" >}} driver. In order to use this
 feature, the
 [Ceph CSI driver](https://github.com/ceph/ceph-csi)
-must be installed on the cluster and the `CSIMigration` and `CSIMigrationRBD`
+must be installed on the cluster and the `CSIMigration` and `csiMigrationRBD`
```
Comment (divya-mohan0209) [inline content/en/docs/concepts/storage/volumes.md:861]: Per the [docs style guide](https://kubernetes.io/docs/contribute/style/style-guide/), the current format is accurate and appropriate  since while discussing API objects UpperCamelCase is used.

File: content/en/docs/reference/command-line-tools-reference/feature-gates.md [hunk 2]
```diff
@@ -85,7 +85,7 @@ different Kubernetes components.
 | `CSIMigrationOpenStack` | `true` | Beta | 1.18 | |
 | `CSIMigrationvSphere` | `false` | Beta | 1.19 | |
 | `CSIMigrationPortworx` | `false` | Alpha | 1.23 | |
-| `CSIMigrationRBD` | `false` | Alpha | 1.23 | |
+| `csiMigrationRBD` | `false` | Alpha | 1.23 | |
```
Comment (divya-mohan0209) [inline content/en/docs/reference/command-line-tools-reference/feature-gates.md:88]: Per the [docs style guide](https://kubernetes.io/docs/contribute/style/style-guide/), the current format is accurate and appropriate  since while discussing API objects UpperCamelCase is used.

File: content/en/docs/reference/command-line-tools-reference/feature-gates.md [hunk 3]
```diff
@@ -638,9 +638,9 @@ Each feature gate is designed for enabling/disabling a specific feature:
   operations from the GCE-PD in-tree plugin to PD CSI plugin. Supports falling
   back to in-tree GCE plugin if a node does not have PD CSI plugin installed and
   configured. Requires CSIMigration feature flag enabled.
-- `CSIMigrationRBD`: Enables shims and translation logic to route volume
+- `csiMigrationRBD`: Enables shims and translation logic to route volume
```
Comment (divya-mohan0209) [inline content/en/docs/reference/command-line-tools-reference/feature-gates.md:641]: Per the [docs style guide](https://kubernetes.io/docs/contribute/style/style-guide/), the current format is accurate and appropriate  since while discussing API objects UpperCamelCase is used.

File: content/en/docs/reference/command-line-tools-reference/feature-gates.md [hunk 4]
```diff
@@ -638,9 +638,9 @@ Each feature gate is designed for enabling/disabling a specific feature:
...
   operations from the RBD in-tree plugin to Ceph RBD CSI plugin. Requires
-  CSIMigration and CSIMigrationRBD feature flags enabled and Ceph CSI plugin
+  CSIMigration and csiMigrationRBD feature flags enabled and Ceph CSI plugin
```
Comment (divya-mohan0209) [inline content/en/docs/reference/command-line-tools-reference/feature-gates.md:643]: Per the [docs style guide](https://kubernetes.io/docs/contribute/style/style-guide/), the current format is accurate and appropriate  since while discussing API objects UpperCamelCase is used.

**Review by divya-mohan0209 — COMMENTED** [review]
Thank you for your contributions. However, per the [docs style guide](https://kubernetes.io/docs/contribute/style/style-guide/) the changes suggested are not in line with our styling/formatting standards. Hence closing the PR.
/close

**Comment by tengqm** [comment]
/reopen
This one is special, sounds to me that the feature gate was already incorrectly spelt. We simply need to document the fact.

**Comment by divya-mohan0209** [comment]
@tengqm could you elaborate on what needs to be done here?

**Comment by tengqm** [comment]
@divya-mohan0209 This one should be merged as is, IIUC.

**Comment by divya-mohan0209** [comment]
@tengqm , I agree that there has been a recent change and we need to document it. However, per my understanding in feature gates as well as in other documentation, the styling guide requires us to follow the below guideline.

> When you refer specifically to interacting with an API object, use UpperCamelCase, also known as Pascal case. You may see different capitalization, such as "configMap", in the API Reference. When writing general documentation, it's better to use upper camel case, calling it "ConfigMap" instead.

When you are generally discussing an API object, use sentence-style capitalization.

Please could we have some clarity as to why this merge will help us understand the change better?

**Comment by tengqm** [comment]
@divya-mohan0209 While the API guideline is crystal clear about API resource kinds (types), the issue we face here is not exactly the same. A feature gate is something we specify from command line, i.e. `--feature-gates <gate name>=True|False`. The CSIMigrationRBD feature gate should have been spelt as `CSIMigrationRBD`. However, due to upstream code implementation problem, it had been incorrectly put as `csiMigrationRBD`. That means if you want to turn off this feature gate, you will specify `--feature-gates csiMigrationRBD=False` rather than `--feature-gates CSIMigrationRBD=False`. This is the "fact" I referred to. It is kind of a bug, but we have to document faithfully how things work.

/hold

@humblec This should be targeted at `main` branch because the fix has been cherry-picked into release 1.23?

**Comment by divya-mohan0209** [comment]
I agree we need to tackle that, but making amendments to the feature gates documentation in the way proposed might not be the best way ahead IMO. Don't mean for my comment to be blocker, however we do need to ensure we're able to document it as faithfully as possible.

**Comment by tengqm** [comment]
@divya-mohan0209 It is not a situation we see everyday. Please also see the [comments in the upstream code review](https://github.com/kubernetes/kubernetes/pull/107571#issuecomment-1013396841). The key is not to break users.

**Comment by divya-mohan0209** [comment]
@tengqm : I agree that we should achieve our key of not breaking users, however I'd also like for this to be discussed in Tuesday's slack standup since in the event of such special cases in future there needs to be a documented procedure or agreement to convey this in a better fashion. Please let me know if that sounds like an acceptable option?

**Comment by humblec** [comment]
> @divya-mohan0209 While the API guideline is crystal clear about API resource kinds (types), the issue we face here is not exactly the same. A feature gate is something we specify from command line, i.e. `--feature-gates <gate name>=True|False`. The CSIMigrationRBD feature gate should have been spelt as `CSIMigrationRBD`. However, due to upstream code implementation problem, it had been incorrectly put as `csiMigrationRBD`. That means if you want to turn off this feature gate, you will specify `--feature-gates csiMigrationRBD=False` rather than `--feature-gates CSIMigrationRBD=False`. This is the "fact" I referred to. It is kind of a bug, but we have to document faithfully how things work.
> 
> /hold

@tengqm You explained it correct!, ideally it would have been `CSIMigrationRBD` , but 1.23 release had it in `csiMigrationRBD` and this PR correct the documentation too. 


> 
> @humblec This should be targeted at `main` branch because the fix has been cherry-picked into release 1.23?

No, it is fixed in main branch, however not to break the `1.23` users, the cherrypick is not done to 1.23 , so 1.23 will keep `csiMigrationRBD` and main branch has fixed it to `CSIMigrationRBD` so doc is matching that.

In short, this doc change atm  only required for 1.23 branch. 

Cc @divya-mohan0209

**Comment by tengqm** [comment]
@humblec The 'dev-1.23' branch is a temporary branch when 1.23 was still under development. The 'main' branch, at the moment, is the official 1.23 documentation. This is the reason why this should target 'main'.

**Comment by humblec** [comment]
> @humblec The 'dev-1.23' branch is a temporary branch when 1.23 was still under development. The 'main' branch, at the moment, is the official 1.23 documentation. This is the reason why this should target 'main'.

Yeah, I was bit in doubt about this temp branch, so opened a PR here by closing  https://github.com/kubernetes/website/pull/31349 , but now the other one has reopened, shall we close this one in that case? Thanks @tengqm

**Comment by tengqm** [comment]
/close
Closing in favor of #31349
